org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/DebugAdapter.java

```diff
@@ -466,13 +472,13 @@ private void handleEvent(DebugEvent debugEvent) {
             ThreadReference deathThread = ((ThreadDeathEvent) event).thread();
             Events.ThreadEvent threadDeathEvent = new Events.ThreadEvent("exited", deathThread.uniqueID());
             this.sendEvent(threadDeathEvent);
-        } else if (event instanceof BreakpointEvent || event instanceof StepEvent) {
-            try {
-                Events.StoppedEvent stopEvent = this.convertDebuggerStoppedEventToClient((LocatableEvent) event);
-                this.sendEvent(stopEvent);
-            } catch (AbsentInformationException | URISyntaxException e) {
-                Logger.logException("Get breakpoint/step info exception", e);
-            }
+        } else if (event instanceof BreakpointEvent) {
+            ThreadReference bpThread = ((BreakpointEvent) event).thread();
+            this.sendEvent(new Events.StoppedEvent("breakpoint", bpThread.uniqueID()));
+            debugEvent.shouldResume &= false;
```

Comment:
Why do we use &= operator? EventHub actually does so.
I think the assignment is enough here.